packages/tools/src/tools/annotation/SplineROITool.ts

```diff
@@ -0,0 +1,895 @@
+import { isAnnotationVisible } from '../../stateManagement/annotation/annotationVisibility';
+import {
+  drawHandles as drawHandlesSvg,
+  drawPolyline as drawPolylineSvg,
+  drawLinkedTextBox as drawLinkedTextBoxSvg,
+} from '../../drawingSvg';
+import { state } from '../../store';
+import { Events, MouseBindings, KeyboardBindings } from '../../enums';
+import { getViewportIdsWithToolToRender } from '../../utilities/viewportFilters';
+import { getTextBoxCoordsCanvas } from '../../utilities/drawing';
+import { resetElementCursor } from '../../cursors/elementCursor';
+import {
+  EventTypes,
+  ToolHandle,
+  TextBoxHandle,
+  PublicToolProps,
+  ToolProps,
+  SVGDrawingHelper,
+} from '../../types';
+import { SplineROIAnnotation } from '../../types/ToolSpecificAnnotationTypes';
+import { AnnotationCompletedEventDetail } from '../../types/EventTypes';
+import triggerAnnotationRenderForViewportIds from '../../utilities/triggerAnnotationRenderForViewportIds';
+import { StyleSpecifier } from '../../types/AnnotationStyle';
+import { ISpline } from './splines/types/ISpline';
+import { CardinalSpline } from './splines/CardinalSpline';
+import { LinearSpline } from './splines/LinearSpline';
+import { CatmullRomSpline } from './splines/CatmullRomSpline';
+import { BSpline } from './splines/BSpline';
+
+const SPLINE_MIN_POINTS = 3;
+const SPLINE_CLICK_CLOSE_CURVE_DIST = 10;
+
+class SplineROITool extends AnnotationTool {
+  static toolName;
+
+  touchDragCallback: any;
+  mouseDragCallback: any;
+  _throttledCalculateCachedStats: any;
+  editData: {
+    annotation: SplineROIAnnotation;
+    viewportIdsToRender: Array<string>;
+    handleIndex?: number;
+    movingTextBox?: boolean;
+    newAnnotation?: boolean;
+    hasMoved?: boolean;
+  } | null;
+  isDrawing: boolean;
+  isHandleOutsideImage = false;
+  splines: Map<string, ISpline> = new Map();
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        preventHandleOutsideImage: false,
+        spline: {
+          configuration: {
+            CARDINAL: {
+              Class: CardinalSpline,
+              scale: 0.5,
+            },
+            CATMULLROM: {
+              Class: CatmullRomSpline,
+            },
+            LINEAR: {
+              Class: LinearSpline,
+            },
+            BSPLINE: {
+              Class: BSpline,
+              addControlPointEnabled: false,
+              deleteControlPointEnabled: false,
+              showControlPointsConnectors: true,
+            },
+            DEFAULT: {
+              resolution: 20,
+              addControlPointLineDistance: 6,
+              delControlPointDistance: 6,
+              addControlPointEnabled: true,
+              deleteControlPointEnabled: true,
+              showControlPointsConnectors: false,
+            },
+          },
+          type: 'CATMULLROM',
```

Comment:
different types can be static prop on the tool so that in the example or app SplineROITool.CATMULLROM

Reply:
Updated